### PR TITLE
lifecycle tests: check no legacy tags on shared resources

### PR DESCRIPTION
**Pull request by justinsb**

### CODE REVIEW

**Comment by justinsb** [comment]
As we discussed @chrislovecnm

File: cmd/kops/lifecycle_integration_test.go [hunk 1]
```diff
@@ -217,6 +219,9 @@ func runLifecycleTest(h *testutils.IntegrationTestHarness, o *LifecycleTestOptio
 				if ownership != expect {
 					t.Errorf("unexpected kubernetes.io/cluster/ tag on %q: actual=%q expected=%q", id, ownership, expect)
 				}
+				if legacy != "" {
+					t.Errorf("added (legacy) KubernetesCluster tag on %q, but it is shared", id)
```
Comment (chrislovecnm) [inline cmd/kops/lifecycle_integration_test.go]: Well, shouldn't we do this on route tables?

Comment (justinsb) [reply cmd/kops/lifecycle_integration_test.go]: Can you say why you think so?

**Pull request merged**